sync/src/block/extension.rs

```diff
@@ -267,28 +267,36 @@ impl NetworkExtension for Extension {
                 }
             }
             SYNC_EXPIRE_TOKEN_BEGIN...SYNC_EXPIRE_TOKEN_END => {
-                let mut requests = self.requests.write();
-                let mut tokens_info = self.tokens_info.write();
-                let token_info = tokens_info.get_mut(&token).unwrap();
-                if token_info.request_id.is_none() {
-                    return
-                }
+                let (id, request_id) = {
+                    let mut tokens_info = self.tokens_info.write();
+                    let token_info = tokens_info.get_mut(&token);
+                    if token_info.is_none() {
+                        return
+                    }
 
-                let id = token_info.id;
-                let request_id = token_info.request_id.unwrap();
-                let request_list = requests.get_mut(&id).unwrap();
+                    let token_info = token_info.unwrap();
+                    if token_info.request_id.is_none() {
+                        return
+                    }
+                    let id = token_info.id;
+                    let request_id = token_info.request_id.unwrap();
+                    token_info.request_id = None;
+                    (id, request_id)
+                };
 
-                let expired_request = request_list.iter().find(|(r, _)| *r == request_id).cloned();
-                if let Some((request_id, request)) = expired_request {
-                    match request {
-                        RequestMessage::Bodies(hashes) => {
-                            self.body_downloader.lock().reset_downloading(&hashes);
+                if let Some(requests) = self.requests.write().get_mut(&id) {
+                    let expired_request = requests.iter().find(|(r, _)| *r == request_id).cloned();
```

Comment:
It's not about this patch, but there is no reason to call `cloned()`.

Reply:
Fixed